app/models/grda_warehouse/tasks/scrub_pii/scrub_all_pii_task.rb

```diff
@@ -0,0 +1,94 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module GrdaWarehouse::Tasks::ScrubPii
+  # Scrub personally identifiable information (PII) from all warehouse and reporting tables
+  class ScrubAllPiiTask
+    def self.perform(...)
+      new.perform(...)
+    end
+
+    def perform(...)
+      with_lock do
+        total = 0
+        scrubber = Pii::Scrubber::ScrubModelPii.new(...)
+        models.each do |model|
+          total += scrubber.perform(model.unscoped)
+        end
+        total
+      end
+    end
+
+    protected
+
+    def models
+      [
+        # misc
+        TextMessage::TopicSubscriber,
+        GrdaWarehouse::HealthEmergency::UploadedTest,
+        GrdaWarehouse::Contact::Base,
+        GrdaWarehouse::Hmis::Staff,
+        GrdaWarehouse::HmisClient,
+        # warehouse.ClientUnencrypted, # no model?
+
+        # hud records
+        GrdaWarehouse::Hud::Client,
+        GrdaWarehouse::Hud::User,
```

Comment:
This feels a bit out of place. All the other records are scrubbing client data, whereas this is scrubbing user data. We are not clearing out the `users` table, so why bother with the hud User. I dont _think_ it would be a problem, just seems different from the other pii here

Reply:
I _think_ we want that here, but definitely understand that this is not Client PII, but someone else's PII.  Similar to client contacts, or case managers.  It is an interesting question if we were scrubbing PII for a single client, would we also scrub the associated `User` records?

Reply:
Ok, I see yes this right for scrubbing *all* PII. 

> It is an interesting question if we were scrubbing PII for a single client, would we also scrub the associated User records?

I would say no, because there are other Clients/records that are referencing that User. We don't want to lose data about other clients.